spec/system/distribution_system_spec.rb

```diff
@@ -0,0 +1,235 @@
+      select @storage_location.name, from: "From storage location"
+
+      fill_in "Comment", with: "Take my wipes... please"
+
+      expect do
+        click_button "Save", match: :first
+      end.to change { PartnerMailerJob.jobs.size }.by(1)
+
+      expect(page).to have_content "Distributions"
+      expect(page.find(".alert-info")).to have_content "reated"
+    end
+  end
+
+  it "User doesn't fill storage_location" do
+    visit @url_prefix + "/distributions/new"
+
+    select @partner.name, from: "Partner"
+    select "", from: "From storage location"
+
+    click_button "Save", match: :first
+    expect(page).to have_content "An error occurred, try again?"
+  end
+
+  context "With an existing distribution" do
+    let!(:distribution) { create(:distribution, :with_items, agency_rep: "A Person", organization: @user.organization) }
+
+    before do
+      visit @url_prefix + "/distributions"
+    end
+
+    it "the user can make changes to it" do
+      click_on "Edit", match: :first
+      expect do
+        fill_in "Agency representative", with: "SOMETHING DIFFERENT"
+        click_on "Save", match: :first
+        distribution.reload
+      end.to change { distribution.agency_rep }.to("SOMETHING DIFFERENT")
+    end
+
+    it "the user can reclaim it" do
+      expect do
+        accept_confirm do
+          click_on "Reclaim"
+        end
+        page.find ".alert"
+      end.to change { Distribution.count }.by(-1)
+      expect(page).to have_content "reclaimed"
+    end
+
+    context "when one of the items has been 'deleted'" do
+      it "the user can still reclaim it and it reactivates the item" do
+        item = distribution.line_items.first.item
+        item.destroy
+        expect do
+          accept_confirm do
+            click_on "Reclaim"
+          end
+          page.find ".alert"
+        end.to change { Distribution.count }.by(-1).and change { Item.count }.by(1)
+        expect(page).to have_content "reclaimed"
+      end
+    end
+  end
+
+  context "When creating a distribution and items have value" do
+    before do
+      item1 = create(:item, value: 10.5)
+      item2 = create(:item)
+      item3 = create(:item, value: 1)
+      @distribution1 = create(:distribution, :with_items, item: item1, agency_rep: "A Person", organization: @user.organization)
+      create(:distribution, :with_items, item: item2, agency_rep: "A Person", organization: @user.organization)
+      @distribution3 = create(:distribution, :with_items, item: item3, agency_rep: "A Person", organization: @user.organization)
+      visit @url_prefix + "/distributions"
+    end
+
+    it 'the user sees value in row on index page' do
+      # row: 100 items * 1$
+      expect(page).to have_content "$100"
+    end
+
+    it 'the user sees total value on index page' do
+      # 100 items * 10.5 + 100 items * 1
+      expect(page).to have_content "$1,150"
+    end
+
+    it 'the user sees value per item on show page' do
```

Comment:
This one is going to fail because the view has changed, you can delete it or skip it
```suggestion
    xit 'the user sees value per item on show page' do
```